main.tf

```diff
@@ -276,32 +280,87 @@ locals {
+      terraform_smart_sanitization     = try(local.stack_configs[stack].terraform_smart_sanitization, var.terraform_smart_sanitization)
+      terraform_version                = try(local.stack_configs[stack].terraform_version, var.terraform_version)
+      worker_pool_id                   = try(local.stack_configs[stack].worker_pool_id, var.worker_pool_id)
+
+      # AWS Integration properties
+      aws_integration_id = try(local.stack_configs[stack].aws_integration_id, var.aws_integration_id)
+
+      # Drift detection properties
+      drift_detection_ignore_state = try(local.stack_configs[stack].drift_detection_ignore_state, var.drift_detection_ignore_state)
+      drift_detection_reconcile    = try(local.stack_configs[stack].drift_detection_reconcile, var.drift_detection_reconcile)
+      drift_detection_schedule     = try(local.stack_configs[stack].drift_detection_schedule, var.drift_detection_schedule)
+      drift_detection_timezone     = try(local.stack_configs[stack].drift_detection_timezone, var.drift_detection_timezone)
+
+      # Destructor properties
+      destructor_deactivated = try(local.stack_configs[stack].destructor_deactivated, var.destructor_deactivated)
+    }
+  }
+
+  # Helper for array merging with fallback
+  stack_array_merger = {
+    for stack in local.stacks : stack => {
+      after_apply    = compact(concat(try(local.stack_configs[stack].after_apply, []), var.after_apply))
+      after_destroy  = compact(concat(try(local.stack_configs[stack].after_destroy, []), var.after_destroy))
+      after_init     = compact(concat(try(local.stack_configs[stack].after_init, []), var.after_init))
+      after_perform  = compact(concat(try(local.stack_configs[stack].after_perform, []), var.after_perform))
+      after_plan     = compact(concat(try(local.stack_configs[stack].after_plan, []), var.after_plan))
+      after_run      = compact(concat(try(local.stack_configs[stack].after_run, []), var.after_run))
+      before_apply   = compact(coalesce(try(local.stack_configs[stack].before_apply, []), var.before_apply))
+      before_destroy = compact(coalesce(try(local.stack_configs[stack].before_destroy, []), var.before_destroy))
+      before_perform = compact(coalesce(try(local.stack_configs[stack].before_perform, []), var.before_perform))
+      before_plan    = compact(coalesce(try(local.stack_configs[stack].before_plan, []), var.before_plan))
+    }
+  }
+
+  # Generic stack filtering helper
+  filtered_stacks = {
+    aws_integration = {
+      for stack, config in local.stack_configs :
+      stack => config if try(config.aws_integration_enabled, var.aws_integration_enabled)
+    }
+    drift_detection = {
+      for stack, config in local.stack_configs :
+      stack => config if try(config.drift_detection_enabled, var.drift_detection_enabled)
+    }
+    destructor = {
+      for stack, config in local.stack_configs :
+      stack => config if try(config.destructor_enabled, var.destructor_enabled)
+    }
+  }
```

Comment:
This is not saving us anything -- it's just moving the logic around. I'd revert this particular bit.

Reply:
Agree on this.